### PR TITLE
Python client contributing

### PR DESCRIPTION
**Pull request by kindyluv**

### CODE REVIEW

File: client/python/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+<hr />
+
+Armada Project welcomes contributions from the opensource community.
+### Note that you need PYTHON 3+ to get this project working and this project does not work on native windows OS.
+
+## Setting up your own fork of this repository on linux, Mac, and WSL OS.
+<hr />
+
+1) On github interface click on the `Fork` button.
+2) Clone your forked of armada repository on your personal git repository by running git clone `github.com:Your_Git_Username/armada.git`.
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:12]:
```suggestion
2) Clone your forked of armada repository on your personal git repository by running `git clone github.com:Your_GitHub_Username/armada.git`.
```

File: client/python/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+How to develop on this python client project 
+=
+<hr />
+
+Armada Project welcomes contributions from the opensource community.
+### Note that you need PYTHON 3+ to get this project working and this project does not work on native windows OS.
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:6]:
```suggestion
This client is written primarily targeting supported versions of python 3 on Linux, Mac OS, and WSL on Windows. 
```

I suggest we make this more declarative, we can't list *all* the things we don't support so lets just list the things we do support.

File: client/python/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Setting up your own fork of this repository on linux, Mac, and WSL OS.
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:8]:
```suggestion
## Setting up your own fork of this repository on Linux, Mac, or WSL OS.
```

File: client/python/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+<hr />
+
+1) On github interface click on the `Fork` button.
+2) Clone your forked of armada repository on your personal git repository by running git clone `github.com:Your_Git_Username/armada.git`.
+3) Enter the directory `cd armada`
+4) Add upstream repository [Armada Repo](https://github.com/G-Research/armada.git)
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:14]:
```suggestion
4) Add upstream repository `git remote add upstream https://github.com/G-Research/armada.git`
```

File: client/python/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+## Install Pyenv/Python and Poetry
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:16]: We're about to merge a change in how the client is built, which invalidates some of this thru no fault of your own.

```suggestion
## Install Pyenv/Python
```

File: client/python/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+<hr />
+
+2) Install [Pyenv](https://github.com/pyenv/pyenv) by following the instructions [here](https://realpython.com/intro-to-pyenv/).
+3) Install [poetry](https://python-poetry.org/) on your local machine by following the instructions on [python poetry documentation](https://python-poetry.org/docs/).
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:20]: This will no longer be required

File: client/python/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+## Setting up your own environment
+<hr />
+
+1) Run `make python` then `cd client/python` and run `poetry install` to pull all necessary dependencies.
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:25]:
```suggestion
1) Run `make python` then `cd client/python` and run `pip install .` to install the client and all dependencies. You need to re-run the pip install command as you make changes.
```

File: client/python/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+## Run test to ensure everything is working
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:27]: We should encourage people to run tests via tox.

`tox`, from the root of client/python, should be all people need to run to get unit tests and formatting run.

File: client/python/README.md [hunk 9]
```diff
@@ -1,19 +1,26 @@
-# Armada Python client
+Armada Python Client
+=
+<hr />
 
+## How To Use This Application
+***
+
+> **Do Not Fork** this application is meant to be used from **[armada_client](https://pypi.org/project/armada-client/)**
```
Comment (jayofdoom) [inline client/python/README.md:8]: Alternatively, this line could be completely removed.

```suggestion
These instructions are intended for end-users. If you wish to develop against armada_client, please see our [documentation on contributing](CONTRIBUTING.md)
```

(and obviously if you took this suggestion, you'd remove the contributing link below.

File: client/python/README.md [hunk 10]
```diff
@@ -1,19 +1,26 @@
...
 
 
 ## Build
 Prerequisites:
 
 For building the python client:
-In the repo level, run `make python`
-`cd client/python`
-`poetry install`
+In the repo level, run `make python` then `cd client/python` and lastly run `poetry install` to pull all the necessary dependencies.
```
Comment (jayofdoom) [inline client/python/README.md:23]:
```suggestion
In the repo level, run `make python` then `cd client/python` and lastly run `pip install .` to install the client from git with all necessary dependencies.
```

File: client/python/README.md [hunk 11]
```diff
@@ -22,7 +29,7 @@ In the repo level, run `make python`
     - Package is defined by pyproject.toml
     - `poetry install` will pull dependencies and install based on pyproject.toml
 3) tox
-    - `poetry run tox -e format` will check formating/linter for your code according to default black settings and flake8
+    - `poetry run tox -e format` will check formatting/linter for your code according to default black settings and flake8
```
Comment (jayofdoom) [inline client/python/README.md:32]: This whole second will need some de-poetry-ing love. I think you can replace `poetry install` on line 30 with `pip install .`, and just remove the `poetry run` prefix from all these tox commands.

**Review by jayofdoom — COMMENTED** [review]
Sorry if the comments about getting rid of poetry from these docs are a little out of scope, but writing this doc targeting the generic tool (pip) will allow it to work after the setuptools migration is merged.

File: client/python/CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
+How to develop on this python client project 
+=
+<hr />
+
+Armada Project welcomes contributions from the opensource community.
+### This client is written primarily targeting supported versions PYTHON 3 on Linux, Mac, and WSL on Windows.
```
Comment (jayofdoom) [inline client/python/CONTRIBUTING.md:6]:
```suggestion
This client is written primarily targeting supported versions of python 3 on Linux, Mac, and WSL on Windows.
```
Just making this no longer a header, and cleaning up the wording

File: client/python/README.md (hunk 9 above)
Comment (jayofdoom) [reply client/python/README.md:8]: @kindyluv doesn't look like this comment has been addressed

File: client/python/README.md (hunk 11 above)
Comment (jayofdoom) [reply client/python/README.md:32]: @kindyluv doesn't look like this comment has been addressed

**Review by jayofdoom — COMMENTED** [review]
Given our conversation in slack covered changes to the README.md file; I presume those changes might not have been pushed up? None of my review comments in that file have been addressed yet. Thanks for the update on the other file!

File: client/python/README.md [hunk 13]
```diff
@@ -1,31 +1,39 @@
...
-    - Sets up local python environment for supporting multiple python environments
-    - Set up a local python 3.9 environment
-2) poetry
-    - Package is defined by pyproject.toml
-    - `poetry install` will pull dependencies and install based on pyproject.toml
+   - Sets up local python environment for supporting multiple python environments
+   - Set up a local python 3.9 environment
+2) pip
+   - Package is defined by pyproject.toml
+   - `pip install` will pull dependencies and install based on pyproject.toml
```
Comment (jayofdoom) [inline client/python/README.md:31]: This needs the `.`, `pip install .`

**Review by jayofdoom — COMMENTED** [review]
Super close! Just need to address those two outstanding comments.

**Review by jayofdoom — DISMISSED** [review]

File: client/python/README.md [hunk 14]
```diff
@@ -1,31 +1,39 @@
...
 
 
 ## Build
 Prerequisites:
 
 For building the python client:
-In the repo level, run `make python`
-`cd client/python`
-`poetry install`
+In the repo level, run `make python` then `cd client/python` and lastly run `pip install` to install the client from git with all necessary dependencies.
```
Comment (kannon92) [inline client/python/README.md:24]:
```suggestion
In the repo level, run `make python` then `cd client/python` and lastly run `pip install .` to install the client from git with all necessary dependencies.
```

Comment (jayofdoom) [reply client/python/README.md:24]: @kindyluv if this comment is addressed (you can just hit the commit suggestion button) this PR will become mergeable

**Review by jayofdoom — APPROVED** [review]

**Pull request merged**